custom_components/unifiprotect/__init__.py

```diff
@@ -110,17 +208,46 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
         protect_data=protect_data,
         protect=protect,
         disable_stream=entry.options.get(CONF_DISABLE_RTSP, False),
-        doorbell_text=entry.options.get(CONF_DOORBELL_TEXT, None),
     )
 
     await _async_get_or_create_nvr_device_in_registry(hass, entry, nvr_info)
 
-    hass.config_entries.async_setup_platforms(entry, PLATFORMS)
+    if above_ha_version(2021, 12):
+        hass.config_entries.async_setup_platforms(entry, PLATFORMS_NEXT)
+    else:
+        hass.config_entries.async_setup_platforms(entry, PLATFORMS)
+
+    async def add_doorbell_text(call: ServiceCall):
+        message: str = call.data[CONF_MESSAGE]
+        await protect.bootstrap.nvr.add_custom_doorbell_message(message)
+
+    async def remove_doorbell_text(call: ServiceCall):
+        message: str = call.data[CONF_MESSAGE]
+        await protect.bootstrap.nvr.remove_custom_doorbell_message(message)
+
+    async def set_default_doorbell_text(call: ServiceCall):
+        message: str = call.data[CONF_MESSAGE]
+        await protect.bootstrap.nvr.set_default_doorbell_message(message)
```

Comment:
These should be entity services since they can have multiple protect installs

https://developers.home-assistant.io/docs/dev_101_services/#entity-services